### PR TITLE
Bump java deps

### PR DESCRIPTION
**Pull request by charliestrawn-wf**
### Story:
In order to not bombard the repo with PRs, I rolled up the dependency bumps that dependabot will make for java. I closed them on my personal fork as I went through them as you can see [here](https://github.com/charliestrawn/frugal/pulls?q=is%3Apr+is%3Aclosed+label%3Ajava+label%3Adependencies). In order to get the build to pass I had to disable failing on checkstyle violations. I'm not sure if these were just warnings before or if they are just new style guidelines, but I think they probably warrant a follow up PR.

### Acceptance Criteria:
- [ ] At least one InfRe Squad 2 member has reviewed and +1'd
- [ ] Code has been tested and results documented
- [ ] Unit tests have been updated
- [ ] Updates to documentation if necessary
- [ ] Verify and document changes to any other Messaging components
- [ ] Pull request made against the 'develop' branch, not master

#### Reviewers:
@Workiva/product2

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
<details>
<summary>(9) Vulnerable direct dependencies (fix required by april 1, 2020) were detected</summary>
<li>1 vulns in <code>io.netty:netty-all < 4.1.42</code> via <code>examples/java/pom.xml</code></li>
<li>1 vulns in <code>io.netty:netty-all < 4.1.42</code> via <code>lib/java/pom.xml</code></li>
<li>1 vulns in <code>io.netty:netty-all < 4.1.42</code> via <code>test/integration/java/frugal-integration-test/pom.xml</code></li>
<li>1 vulns in <code>org.apache.activemq:activemq-client < 5.15.9</code> via <code>examples/java/pom.xml</code></li>
<li>1 vulns in <code>org.apache.activemq:activemq-client < 5.15.9</code> via <code>test/integration/java/frugal-integration-test/pom.xml</code></li>
<li>2 vulns in <code>com.puppycrawl.tools:checkstyle < 8.29</code> via <code>lib/java/pom.xml</code></li>
<li>2 vulns in <code>org.apache.thrift:libthrift < 0.12.0</code> via <code>examples/java/pom.xml</code></li>
<li>2 vulns in <code>org.apache.thrift:libthrift < 0.12.0</code> via <code>lib/java/pom.xml</code></li>
<li>2 vulns in <code>org.apache.thrift:libthrift < 0.12.0</code> via <code>test/integration/java/frugal-integration-test/pom.xml</code></li>
</details>

## Action Items
* Review dependencies for available updates
* See <a href="https://splunk.workiva.net/en-US/app/search/security_vulnerable_dependencies?form.repository=frugal">this Splunk dashboard</a> for more CVE details
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/java/pom.xml [hunk 1]
```diff
@@ -71,35 +71,35 @@
...
             <artifactId>netty-all</artifactId>
-            <version>4.1.6.Final</version>
+            <version>4.1.42.Final</version>
             <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>javax.servlet</groupId>
             <artifactId>javax.servlet-api</artifactId>
-            <version>3.1.0</version>
+            <version>4.0.1</version>
```
Comment (stevenosborne-wf) [inline lib/java/pom.xml]: Major version change - test locally

File: lib/java/pom.xml [hunk 2]
```diff
@@ -165,23 +165,24 @@
...
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>2.19.1</version>
+                <version>2.22.2</version>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-checkstyle-plugin</artifactId>
-                <version>2.17</version>
+                <version>3.1.0</version>
```
Comment (stevenosborne-wf) [inline lib/java/pom.xml]: Major version change - but just a build plugin - ok if build passes.

File: lib/java/pom.xml [hunk 3]
```diff
@@ -165,23 +165,24 @@
...
                 <dependencies>
                     <dependency>
                         <groupId>com.puppycrawl.tools</groupId>
                         <artifactId>checkstyle</artifactId>
-                        <version>6.15</version>
+                        <version>8.29</version>
```
Comment (stevenosborne-wf) [inline lib/java/pom.xml]: Major version change - but just a build plugin - ok if build passes.

File: lib/java/pom.xml [hunk 4]
```diff
@@ -276,7 +278,7 @@
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-source-plugin</artifactId>
-                        <version>2.4</version>
+                        <version>3.2.1</version>
```
Comment (stevenosborne-wf) [inline lib/java/pom.xml]: Major version change - but just a build plugin - ok if build passes.

File: test/integration/java/frugal-integration-test/pom.xml [hunk 5]
```diff
@@ -120,7 +120,7 @@
             <plugin>
                 <groupId>org.codehaus.mojo</groupId>
                 <artifactId>build-helper-maven-plugin</artifactId>
-                <version>1.10</version>
+                <version>3.0.0</version>
```
Comment (stevenosborne-wf) [inline test/integration/java/frugal-integration-test/pom.xml]: Major version change - but just a build plugin - ok if build passes.

File: lib/java/pom.xml (hunk 1 above)
Comment (stevenosborne-wf) [reply lib/java/pom.xml]: Good to go here.

**Comment by yuanmwang-wf** [comment]
@Workiva/release-management-p

**Pull request merged**